src/io/flutter/devtools/DevToolsManager.java

```diff
@@ -224,16 +225,20 @@ public void processTerminated(@NotNull ProcessEvent event) {
   }
 
   public void openBrowserAndConnect(String serviceProtocolUri, String page) {
+    final String ideParam = "ide=" + (FlutterUtils.isAndroidStudio() ? "AndroidStudio&" : "IntelliJ&");
     if (serviceProtocolUri == null) {
-      BrowserLauncher.getInstance().browse("http://" + devtoolsHost + ":" + devtoolsPort + "/?hide=debugger&", null);
+      BrowserLauncher.getInstance().browse(
+        "http://" + devtoolsHost + ":" + devtoolsPort + "/?" + ideParam,
+        null
+      );
     }
     else {
       try {
         final String urlParam = URLEncoder.encode(serviceProtocolUri, "UTF-8");
         final String pageParam = page == null ? "" : ("#" + page);
 
         BrowserLauncher.getInstance().browse(
-          "http://" + devtoolsHost + ":" + devtoolsPort + "/?uri=" + urlParam + pageParam,
+          "http://" + devtoolsHost + ":" + devtoolsPort + "/?" + ideParam + "uri=" + urlParam + pageParam,
```

Comment:
create a list of params and join instead of manually adding `&`
That will avoid bugs where we end up with extra of missing `&`

Add a unit test for the URL generation code. It should be pretty easy to add a test for this which would catch the trailing `&` case.